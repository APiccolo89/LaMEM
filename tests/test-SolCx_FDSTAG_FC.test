EXEC            = ../bin/LaMEM -Benchmark_SolCx -mumax 1 -nnode_x 9 -nnode_y 3 -nnode_z 9 -vpt_element FDSTAG -StokesSolver 3
NPROCS          = 1
TEST_TYPE       = tolerance_diff_pl,1.0e-4
