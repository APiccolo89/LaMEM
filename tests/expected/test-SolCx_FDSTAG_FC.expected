# -------------------------------------------------------------------------- 
#                   Lithosphere and Mantle Evolution Model                   
#              Current Revision: 4812 - 2013-10-01 23:57:18		     
#  Modified items: LaMEM_Particles.c, Elements.c, Assembly_FDSTAG.c, fdstag/JacRes.c, 
#     Compiled: Date: Oct  9 2013 - Time: 23:02:54 - Mode:  	    
# -------------------------------------------------------------------------- 
 Stokes solver             : Fully Coupled
 Velocity solver           : Direct (Build-in direct solver)
 Input is assumed to be in non-dimensional units 
 Element-type used         : FDSTAG;  [Q1P0 Q2Pm1_global Q2Pm1_local Q1Q1 FDSTAG] (change with -vpt_element)
 Total # of cpu's          : 1 
 Total # of velocity dof's : 729 
 Computing mean material per element using arithmetic averaging from integration points 
 Employing code in an Eulerian manner 
 Gravity angle with z-axis =  0 
 Approximate Aspect ratio of cells [dx/dz, dy/dz] = [1 4] 
 Number of tracers/cell in [x,y,z]-direction = [2,2,2] 
 NumSurfaceNodes=1  nz=9 
 # Not employing nonlinear iterations 
 BC employed: BC.[Left Right; Front Back; Lower Upper]=[1 1; 1 1; 1 1] 
# Initial time step: 0.001 years 
 Input is assumed to be in non-dimensional units 
 Fine Grid parameters, nel=[8,2,8], cpu=[1,1,1] 
# Finished initializing Stokes matrices : 0.00269485 s 
# Not adding random noise to initial particle distribution. 
Initialized particles 
#rank=0, Attributing 1024 particles to elements took 0.000843048 s 
The total # of particles present is 1024 
 *** Using a falling block setup *** 
 Falling block coordinates:  [left,right]=[0.250000,0.750000]; [front,back]=[0.000000,0.500000]; [bottom,top]=[0.250000,0.750000] 
Saved initial particle distribution to directory ./InitialParticles 
The total memory used by integration points is 3.379200e-02 MB
The max memory used by integration points is 3.379200e-02 MB
The min memory used by integration points is 3.379200e-02 MB
#  Computing material properties from particles for FDSTAG ...   [0.00129414 s] 
#  Interpolating meshes and temperatures took 0.00139809 s 


 --------------------------------------------------------------------------------  
 
             PERFORMING BENCHMARK OF LaMEM VERSUS ANALYTICAL SOLUTION              
  using:                             SolCx                                        
 
   with:      rho   = -1*sin(1*PI*z)*cos(1*PI*x)                                  
              eta   = 1 if x<x_c & eta=eta_b for x>x_c                            
    and:      x_c   = 0.5     (change with -SolCx_xc ... )                         
              eta_b = 1     (change with -mumax ...    )                          

 --------------------------------------------------------------------------------  
 
 
#  Forming FD stiffness matrix mx,my,mz=(9, 3, 9) ...  
#  FDSTAG: Time to insert stencil into operators 4.6301e-04 
#  FDSTAG: Time to insert bc's into operators 4.3154e-05 
#  FDSTAG: Time to assemble operators 4.2915e-05 
#  Forming stiffness matrixes took 0.000804901 s 
FC: Viscosity structure info: min(0)=1.000000e+00 : max(0)=1.000000e+00 : contrast=1.000000e+00 
Using unpreconditioned (true) Stokes residual to define the FC stopping condition 
StokesResidual: Using L_inf 
Stopping conditions: 
Velocity: RTOL: 1.000000e-06	ATOL: 1.000000e-16	minATOL: 1.000000e-05		fc_ksp_min_it 4
Pressure: RTOL: 1.000000e-06	ATOL: 1.000000e-16	minATOL: 1.000000e-05		fc_ksp_min_it 4
StokesResidual: Monitor activated 
# FC: using nonzero initial guess 
  Residual norms for fc_ solve.
  # KSP Iterations      /  L2_norm             scaled_L2_norm      Linf_norm           /  ... for each   2 blocks  [fc_] 
  0 KSP Residual norms  /  5.548159381289e+00  2.054873844922e-01  9.619397662556e-01  /  0.000000000000e+00  0.000000000000e+00  0.000000000000e+00  /  [fc_] 
  1 KSP Residual norms  /  3.536230838511e-02  1.309715125374e-03  6.131116344091e-03  /  4.415260243253e-01  3.902575573384e-02  7.508085269712e-02  /  [fc_] 
  2 KSP Residual norms  /  2.353913360364e-14  8.718197630976e-16  8.942499518660e-15  /  1.493966264020e-15  1.320492095191e-16  4.007211229506e-16  /  [fc_] 
  3 KSP Residual norms  /  8.723740866570e-15  3.231015135767e-16  1.512971171068e-15  /  8.012668312602e-16  7.082265124049e-17  1.601337774795e-16  /  [fc_] 
  4 KSP Residual norms  /  8.723739851451e-15  3.231014759797e-16  1.512521846773e-15  /  7.843024031374e-16  6.932319346992e-17  1.336966230559e-16  /  [fc_] 
Velocity convergence reason: RTOL 
Pressure convergence reason: RTOL 
Linear solve converged due to CONVERGED_HAPPY_BREAKDOWN iterations 4
#  Finished Stokes solve after 0.00300694 s 
------------------------------------------
Stokes solver summary: 
  divergence: 
    Div_min = 0.000000000000e+00 
    Div_max = 1.526556658860e-16 
    |Div|_2 = 8.282588344295e-16 
    |Div|_1 = 7.429398038181e-15 
  momentum: 
    mRes_min = 0.000000000000e+00 
    mRes_max = 4.829470157119e-15 
    |mRes|_2 = 1.555476035159e-14 
    |mRes|_1 = 1.684819739454e-13 
  velocity: 
    u_min   = -2.468177965731e-02 
    u_max   = 2.468177965731e-02 
    |u|_2   = 2.013225944595e-01 
    |u|_1   = 2.593970370555e+00 
  pressure: 
    p_min   = -1.511249304982e-01 
    p_max   = 1.511249304982e-01 
    |p|_2   = 8.887164615523e-01 
    |p|_1   = 8.255564597116e+00 
------------------------------------------
The total memory used by integration points is 3.379200e-02 MB
The max memory used by integration points is 3.379200e-02 MB
The min memory used by integration points is 3.379200e-02 MB
#  Solving for velocity took 0.00413394 s 
The total memory used by integration points is 3.379200e-02 MB
The max memory used by integration points is 3.379200e-02 MB
The min memory used by integration points is 3.379200e-02 MB


 --------------------------------------------------------------------------------  

  LaMEM STOKES BENCHMARK RESULTS:                                                 

       dx = 0.125        [node spacing in x-direction   ]
       dy = 0.5        [node spacing  in y-direction   ] 
       dz = 0.125        [node spacing  in z-direction   ] 

  |Vx |_1 = 0.000242605 
  |Vy |_1 = 4.72304e-19 
  |Vz |_1 = 0.000242605 

  |u  |_1 = 0.00048521   [velocity                      ]
  |P  |_1 = 0.000841833   [pressure                      ]
  |e  |_1 = 0   [strain rate                   ]
  |P  |_2 = 1.05122e-06   [pressure                      ]

 --------------------------------------------------------------------------------  

The total memory used by integration points is 3.379200e-02 MB
The max memory used by integration points is 3.379200e-02 MB
The min memory used by integration points is 3.379200e-02 MB
#  Forming FD stiffness matrix for energy equation with mx,my,mz=(9, 3, 9) with Top/Bottom T=[0,1]...  
#  Forming temperature stiffness matrixes took 7.70092e-05 s 
#  Solving for temperature took 0.000185013 s 
The total memory used by integration points is 3.379200e-02 MB
The max memory used by integration points is 3.379200e-02 MB
The min memory used by integration points is 3.379200e-02 MB
The total memory used by particles is 2.867200e-01 MB
The max memory used by particles is 2.867200e-01 MB
The min memory used by particles is 2.867200e-01 MB
#  Updating properties @ integration points and particles took 0.00198817 s 
AdvectionMethod = 0 
  Starting particle advection 
  Finished particle advection 
Minimum z-level phase 1 0.281238 
# Writing output to existing directory Timestep_000000 
# Saved MATLAB output in file Timestep_000000/AnalyticalBenchmark_SolCx.0.1000000.out 
# Saved Paraview VTS Velocity output in file AnalyticalBenchmark_SolCx_velocity_step000000-mesh.pvts 
# Saved Paraview VTS Temperature output in file AnalyticalBenchmark_SolCx_temperature_step000000-mesh.pvts 
# Saved Paraview VTS Surface Topography output in file AnalyticalBenchmark_SolCx_Surface_topography_step000000-mesh.pvts 
# Saved Paraview VTS Bottom Topography output in file AnalyticalBenchmark_SolCx_Bottom_topography_step000000-mesh.pvts 
# Saved Paraview VTS point-wise quadrature output in file AnalyticalBenchmark_SolCx_quadrature_step000000-mesh.pvts 
 Starting GetParticleNodes 
#rank=0, Attributing 1024 particles to elements took 0.000549078 s 
The total # of particles present is 1024 
 Finished GetParticleNodes 
 Maximum velocity  0.0246818    
 Time = 0.001, dt=0.225087   
# Finished timestep 0 out of 1 in 0.016093 s
  
# Total time required: 0.0216579 s 
#PETSc Option Table entries:
-AddRandomNoiseParticles 0
-Benchmark_SolCx
-StokesSolver 3
-fc_ksp_converged_reason
-fc_ksp_max_it 1000
-fc_ksp_min_it 4
-fc_ksp_p_atol 1.0e-16
-fc_ksp_p_min_atol 1.0e-5
-fc_ksp_p_rtol 1.0e-6
-fc_ksp_type gcr
-fc_ksp_v_atol 1.0e-16
-fc_ksp_v_min_atol 1.0e-5
-fc_ksp_v_rtol 1.0e-6
-fc_pc_fieldsplit_schur_factorization_type UPPER
-fc_pc_fieldsplit_type SCHUR
-fc_pc_type fieldsplit
-mumax 1
-nnode_x 9
-nnode_y 3
-nnode_z 9
-options_left
-ps_ksp_type preonly
-ps_pc_type jacobi
-use_stokes_monitor
-use_stokes_norm Linf
-use_stokes_residual
-vpt_element FDSTAG
-vs_ksp_type preonly
-vs_pc_type lu
#End of PETSc Option Table entries
There are no unused options.
